beprof/curve.py

```diff
@@ -75,8 +78,21 @@ def y(self):
     def y(self, value):
         self[:, 1] = value
 
-    def rescale(self, factor=1.0):
-        self.y /= factor
+    def rescale(self, factor=1.0, allow_cast=True):
+        """
+        todo: write docstring nad doctests
+        :param factor:
+        :param allow_cast:
+        :return:
+        """
+        try:
+            self.y /= factor
+        except TypeError as e:
+            logger.error("Division in place is impossible.\n%s" % e)
+            if allow_cast:
+                self.y = self.y / factor
+            else:
+                logger.error("allow_cast flag set to True should help")
```

Comment:
The thing is - at least for me if something throws exception - it probably did not finish it's job (and possibly rolled back). So raising exception after successful division is... could be misinterpreted?